content/concepts/connections.md

```diff
@@ -0,0 +1,155 @@
+
+An interface called a [switch](/concepts/stream-multiplexing/#switch/swarm) 
+(and sometimes swarm) implements a basic dial queue that manages an application's 
+dialing and listening state. This prevents multiple, 
+simultaneous dial requests to the same peer. 
+
+### How do peers know what transports other peers support?
+
+Each libp2p node has information about the transports it supports.
+Peers can obtain this information by dialing a peer and listen 
+for updates on the network. More information about peers is available 
+on the [peers guide](/concepts/peers).
+
+### How do peers establish a connection once they find others to connect to?
+
+There are different approaches that peers can take to establish a connection.
+The dialing peer, known as the *Initiator* of the connection, sends a dialing request 
+to the listening peer, known as the *Responder* of the connection, via the Initiator's 
+multiaddr over the transport. To accept connections, a libp2p application registers 
+handler functions for protocols using their protocol ID.
+
+More information on multiaddrs, peers, and protocols is available on the
+[addressing](/concepts/addressing), [peers](/concepts/peers),
+[protocol](/concepts/protocols) guides, respectively.
+
+### How does the switch decide on which transport protocol to use?
+
+The switch conducts a protocol negotiation process between two peers through an 
+interface known as [`multistream-select`](https://github.com/multiformats/multistream-select).
+In particular, the multicodec is negotiated between two peers.
+
+### What happens when two peers do not support the same transport?
+
+If a Responder doesn't support a particular protocol, they may respond with "na" 
+(not available), and the Initiator can either try another protocol, request a list 
+of supported protocols, or stop sending dialing requests. If there was an existing 
+connection made to a remote peer, the switch will simply use the existing connection 
+and open another multiplexed stream over it. Stay tuned for a dedicated guide on
+protocol negotiation. For now, the 
+[protocol negotiation section](/concepts/protocols/protocol-negotiation) in the 
+protocol guide outlines the gist of multistream.
+
+Another option is using the [circuit relay](/concepts/circuit-relay) transport 
+protocol, which routes network traffic between two peers over a relay peer. 
+A peer can advertise itself as being reachable through a remote relay node. 
+
+### How can the Responder dial back to the Initiator?
+
+In general, a peer multiaddr is typically discovered with their Peer ID, which
+acts as a unique identifier for each peer in a libp2p network.
+
+If a peer relies on a relay node to listen to a dial request, circuit relaying 
+creates a peer-to-peer circuit by adding the connection path to the connection 
+multiaddr. Thus, a listening peer can use the same path to dial back to a peer 
+that dialed to it.
```

Comment:
This is confusing to me. A listening peer would need to know the local peer's multiaddr to dial back.